stream_buffer.c

```diff
@@ -1204,51 +1197,43 @@ static size_t prvReadBytesFromBuffer( StreamBuffer_t * pxStreamBuffer,
                                       size_t xCount,
                                       size_t xTail )
 {
-    size_t xFirstLength, xNextTail;
+    size_t xFirstLength;
 
-    xNextTail = xTail;
-    if( xCount > ( size_t ) 0 )
-    {
+    configASSERT( xCount != ( size_t ) 0 );

```

Comment:
The non-zero `xCount` check is performed by all callers, so `configASSERT` seems more appropriate here, and lets us drop a level of indentation.

Reply:
Not sure about this one.  It is conceivable a user would pass xBufferLengthBytes as zero into prvReadMessageFromBuffer() at run time - in which case it should recover gracefully.  Also in that case it would err if configASSERT() was not defined.  While it would seem strange to pass xBufferLengthBytes as zero it might happen if, for example, a user passed xBufferLengthBytes as the free space in a buffer and that buffer were to become full.

Reply:
> It is conceivable a user would pass xBufferLengthBytes as zero into prvRead**Message**FromBuffer() at run time

Are we talking about prvRead**Message**FromBuffer() or prvRead**Bytes**FromBuffer()? In either case, the user can't call these private helper functions directly, and any attempt to pass a zero length into these functions is guarded against by existing checks. No guarantees that future code edits preserve all these checks though.